src/Component/Symbolizer/RasterEditor/RasterEditor.tsx

```diff
@@ -57,15 +57,17 @@ import {
   UnsupportedPropertiesContext
 } from '../../../context/UnsupportedPropertiesContext/UnsupportedPropertiesContext';
 import UnsupportedPropertiesUtil from '../../../Util/UnsupportedPropertiesUtil';
-import { useGeoStylerComposition } from '../../../context/GeoStylerContext/GeoStylerContext';
+import { InputConfig, useGeoStylerComposition } from '../../../context/GeoStylerContext/GeoStylerContext';
 
-// default props
-interface RasterEditorDefaultProps {
-  locale: GeoStylerLocale['RasterEditor'];
+export interface RasterEditorComposableProps {
+  opacityField?: InputConfig<OpacityFieldProps['value']>;
+  // TODO add support for default values in ContrastEnhancementField
+  contrastEnhancementField?: Omit<InputConfig<ContrastEnhancementFieldProps['contrastEnhancement']>, 'default'>;
+  gammaValueField?: InputConfig<GammaFieldProps['gamma']>;
 }
 
-// non default props
-export interface RasterEditorProps extends Partial<RasterEditorDefaultProps> {
+export interface RasterEditorInternalProps {
+  locale?: GeoStylerLocale['RasterEditor'];
   contrastEnhancementTypes?: ContrastEnhancement['enhancementType'][];
   symbolizer: RasterSymbolizer;
   onSymbolizerChange?: (changedSymb: Symbolizer) => void;
```

Comment:
`colorRamps` should probably be moved to composable props or removed completely, if it is not used by this component